test/compiler/codegen.jl

```diff
@@ -547,6 +548,11 @@ let a = Core.Intrinsics.trunc_int(UInt24, 3),
     @test sizeof(UInt24) == 3
     @test sizeof(Union{UInt8,UInt24}) == 3
     @test sizeof(Base.RefValue{Union{UInt8,UInt24}}) == 8
+    @test sizeof(Tuple{UInt24}) == 4
+    @test Base.datatype_haspadding(Tuple{UInt24})
+    @test sizeof(UInt136) == 17
+    @test sizeof(Tuple{UInt136}) == 24
```

Comment:
On 32 bit, I'd expect this to only be 20, since the base register size is only 32 bits there. Aligning 136 to the next largest multiple of 32 is 160, or 20 bytes.

Reply:
It should likely reflect the vector register alignment requirements from the ABI, which is likely supposed to be double-word size here (aside from clang bugs).

Reply:
At what size does the double-word alignment kick in?

Reply:
The clang compiler gets the ABI wrong, relative to other compilers and the standard in this. But i128 is supposed to be 128-bit aligned on most 64-bit platforms as an example (such as x86_64)

Reply:
> It should likely reflect the vector register alignment requirements from the ABI, which is likely supposed to be double-word size here (aside from clang bugs).

Are you saying an `i136` should end up 256 bit aligned on x86_64..?

> The clang compiler gets the ABI wrong, relative to other compilers and the standard in this.

How does it get it wrong? On 64 bit, `i128` is supposed to be aligned to 2*64 = 128 already. On 32 bit, it's aligned to 4*32 = 128 as well, if I'm reading the standard right. Is that not what clang is already doing?

Reply:
By not having it set at all, clang kinda gets it right because they hard coded a fix somewhere, but LLVM in general gets it wrong. Fixing it is very easy, but it's an ABI break which is really annoying to handle.

Reply:
We have
```
#  elif defined(_CPU_AARCH64_)
// int128 is 16 bytes aligned on aarch64
#    define MAX_ALIGN 16
```
and 8 on other 64-bit systems. Should we change that?

Reply:
Not in this PR, but we are tracking that in another issue already

Reply:
It's https://github.com/JuliaLang/julia/issues/42326 specifically